src/main/io/serial_cli.c

```diff
@@ -626,12 +639,15 @@ static void cliRxFail(char *cmdline)
                     }
 
                     channelFailsafeConfiguration->step = value;
+                } else if (requireValue) {
+                    cliShowParseError();
+                    return;
                 }
                 channelFailsafeConfiguration->mode = mode;
 
             }
 
-            char modeCharacter = rxFailsafeModes[channelFailsafeConfiguration->mode];
+            char modeCharacter = rxFailsafeModes[channelFailsafeConfiguration->mode + ((channel < NON_AUX_CHANNEL_COUNT) ? 0 : 2)];
```

Comment:
use the `isFlightChannel` variable here too.

Assign the result of the ternary to a variable, called `offset` so you can go `... + offset` to make it clearer.